app/src/lib/stores/git-store.ts

```diff
@@ -523,7 +523,15 @@ export class GitStore extends BaseStore {
     // directory for the next stage. Doing doing a `git checkout -- .` here
     // isn't suitable because we should preserve the other working directory
     // changes.
-    const status = await getStatus(repository)
+
+    const status = await this.performFailableOperation(() =>
+      getStatus(this.repository)
+    )
+
+    if (status == null) {
+      return
```

Comment:
Should we log or report an error here that `undoFirstCommit` couldn't complete because status failed - unlikely as it might be?

Reply:
I think we should log it.

Reply:
I think a `log.warn` here is fine - it'll end up on disk and if we get any new reports of undoing the first commit we'll have the `getStatus` info to work with.